src/OrchardCore/OrchardCore.ContentManagement.Display/ContentItemDisplayManager.cs

```diff
@@ -80,7 +87,27 @@ public async Task<IShape> BuildDisplayAsync(ContentItem contentItem, IUpdateMode
             var metadata = itemShape.Metadata;
             metadata.DisplayType = actualDisplayType;
 
-            // [Stereotype]_[DisplayType]__[ContentType] e.g. Content-BlogPost.Summary
+            if (hasStereotype)
+            {
+                if (actualDisplayType != "Detail")
+                {
+                    // Add fallback/default alternate Stereotype_[DisplayType] e.g. Content.Summary
+                    metadata.Alternates.Add($"Stereotype_{actualDisplayType}");
+
+                    // [Stereotype]_[DisplayType] e.g. Menu.Summary
+                    metadata.Alternates.Add($"{stereotype}_{actualDisplayType}");
```

Comment:
here `{stereotype}_{actualDisplayType}`  == `actualShapeType` you can keep out side if

Reply:
`metadata.Alternates.Add($"{stereotype}_{actualDisplayType}");` is only applicable when `hasStereotype is true` and `actualDisplayType != "Detail"` not sure why we would want to move that out of the if block. And the order of the insertion is important here to allow `{stereotype}_{actualDisplayType}` to take presence when it exists otherwise we fallback on `stereotype_{actualDisplayType}`